src/Compiler/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/InjectDirective.cs

```diff
@@ -69,19 +69,30 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
                 }
 
                 var typeName = tokens[0].Content;
+                var typeSpan = tokens[0].Source;
                 var memberName = tokens[1].Content;
+                var memberSpan = tokens[1].Source;
 
                 if (!properties.Add(memberName))
                 {
                     continue;
                 }
 
-                typeName = typeName.Replace("<TModel>", "<" + modelType + ">");
+                if (typeName.EndsWith("<TModel>", StringComparison.Ordinal))
+                {
+                    typeName = typeName.Replace("<TModel>", "<" + modelType + ">");
+                    if (typeSpan.HasValue)
+                    {
+                        typeSpan = new SourceSpan(typeSpan.Value.FilePath, typeSpan.Value.AbsoluteIndex, typeSpan.Value.LineIndex, typeSpan.Value.CharacterIndex, typeSpan.Value.Length - 8, typeSpan.Value.LineCount, typeSpan.Value.EndCharacterIndex - 8);
```

Comment:
Can we pull those 8's out into a single variable and make them actually `"<TModel>".Length`?